### PR TITLE
Add get_background_items()

### PR DESCRIPTION
**Pull request by olback**
Adding menu entries works.

Callbacks on background menu entries crash, would love some pointers if you know what I'm messing up.

### CODE REVIEW

File: nautilus-extension/src/menu_provider.rs [hunk 1]
```diff
@@ -184,6 +184,28 @@ macro_rules! menu_provider_iface {
...
+                    Some(ref p) => p.get_background_items(window, &file_info),
+                    None => Vec::new(),
+                };
+
+            // dummy top-level Menu for easy recursion
+            let top_menu = Menu {
+                menu_items: file_items,
+            };
+
+            top_menu.to_g_list(Box::into_raw(Box::new(current_folder)) as *mut c_void)
```
Comment (talklittle) [inline nautilus-extension/src/menu_provider.rs]: Thanks for working on this feature!

The crash is caused by this line. It should be:

    top_menu.to_g_list(Box::into_raw(Box::new(file_info)) as *mut c_void)

Comment (olback) [reply nautilus-extension/src/menu_provider.rs]: Thank you!

File: nautilus-extension/src/menu_provider.rs [hunk 2]
```diff
@@ -15,6 +15,7 @@ use translate::file_info_vec_from_g_list;
 
 pub trait MenuProvider : Send + Sync {
     fn get_file_items(&self, window: *mut GtkWidget, files: &Vec<FileInfo>) -> Vec<MenuItem>;
+    fn get_background_items(&self, window: *mut GtkWidget, file: &FileInfo) -> Vec<MenuItem>;
```
Comment (talklittle) [inline nautilus-extension/src/menu_provider.rs]: nitpick: would be nice to name this variable `current_folder` instead of `file`

**Comment by talklittle** [comment]
Anything else you were planning to add, or is this ready to merge?

**Comment by olback** [comment]
Should be ready to merge after a version bump and an update to the changelog.

**Comment by talklittle** [comment]
Thank you very much for the contribution!

**Pull request merged**